autoload/ale/c.vim

```diff
@@ -91,3 +92,27 @@ function! ale#c#FindCompileCommands(buffer) abort
 
     return ''
 endfunction
+
+" Given a buffer number, return the relevant command from
+" compile_commands.json. Return 0 if none could be found.
+function! ale#c#GetCompileCommand(buffer) abort
+    let l:build_dir = ale#c#FindCompileCommands(a:buffer)
+    if empty(l:build_dir)
+        return 0
+    endif
+    let l:commands = json_decode(readfile(l:build_dir . '/compile_commands.json'))
+    let l:file = fnamemodify(bufname(a:buffer), ':p')
+    " save current directory, so we can restore it before returning
+    let l:cwd = getcwd()
```

Comment:
`getcwd()` is banned in the codebase because it doesn't work in asynchronous callbacks, because you could switch to another buffer with a different working directory. Use `expand('#' . a:buffer . ':p')` instead. The Bash script for checking VimL code will mention that in TravisCI.